hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/clustering/plan/strategy/TestSparkConsistentBucketClusteringPlanStrategy.java

```diff
@@ -0,0 +1,190 @@
+import static org.apache.hudi.config.HoodieIndexConfig.BUCKET_MERGE_THRESHOLD;
+import static org.apache.hudi.config.HoodieIndexConfig.BUCKET_SPLIT_THRESHOLD;
+
+public class TestSparkConsistentBucketClusteringPlanStrategy extends HoodieClientTestHarness {
+
+  private final Random random = new Random();
+
+  private void setup() throws IOException {
+    initPath();
+    initSparkContexts();
+    initFileSystem();
+    metaClient = HoodieTestUtils.init(hadoopConf, basePath, HoodieTableType.MERGE_ON_READ);
+  }
+
+  @AfterEach
+  public void tearDown() throws IOException {
+    cleanupResources();
+  }
+
+  @Test
+  public void testBuildSplitClusteringGroup() throws IOException {
+    setup();
+    int maxFileSize = 5120;
+    HoodieWriteConfig config = HoodieWriteConfig.newBuilder().withPath(basePath)
+        .withIndexConfig(HoodieIndexConfig.newBuilder().withIndexType(HoodieIndex.IndexType.BUCKET)
+            .withBucketIndexEngineType(HoodieIndex.BucketIndexEngineType.CONSISTENT_HASHING)
+            .withBucketMaxNum(6)
+            .withBucketNum("4").build())
+        .withStorageConfig(HoodieStorageConfig.newBuilder()
+            .parquetMaxFileSize(maxFileSize).build())
+        .build();
+
+    HoodieTable hoodieTable = HoodieSparkTable.create(config, context, metaClient);
+    SparkConsistentBucketClusteringPlanStrategy planStrategy = new SparkConsistentBucketClusteringPlanStrategy(hoodieTable, context, config);
+
+    HoodieConsistentHashingMetadata metadata = new HoodieConsistentHashingMetadata("partition", config.getBucketIndexNumBuckets());
+    ConsistentBucketIdentifier identifier = new ConsistentBucketIdentifier(metadata);
+
+    int[] fsSize = {maxFileSize * 5, (int) (maxFileSize * BUCKET_SPLIT_THRESHOLD.defaultValue() + 1), maxFileSize, maxFileSize * 5};
+    List<FileSlice> fileSlices = IntStream.range(0, metadata.getNodes().size()).mapToObj(
+        i -> createFileSliceWithSize(metadata.getNodes().get(i).getFileIdPrefix(), 1024, fsSize[i] - 1024)
+    ).collect(Collectors.toList());
+
+    /**
+     * 1. Test split candidate selection based on file size
+     * 2. Test the effectiveness of split slot
+     */
+    Triple res = planStrategy.buildSplitClusteringGroups(identifier, fileSlices, 2);
+    Assertions.assertEquals(2, res.getMiddle());
+    List<HoodieClusteringGroup> groups = (List<HoodieClusteringGroup>) res.getLeft();
+    Assertions.assertEquals(2, groups.size());
+    Assertions.assertEquals(fileSlices.get(0).getFileId(), groups.get(0).getSlices().get(0).getFileId());
+    Assertions.assertEquals(fileSlices.get(1).getFileId(), groups.get(1).getSlices().get(0).getFileId());
+    List<FileSlice> fsUntouched = (List<FileSlice>) res.getRight();
+    Assertions.assertEquals(2, fsUntouched.size());
+    Assertions.assertEquals(fileSlices.get(2), fsUntouched.get(0));
+    Assertions.assertEquals(fileSlices.get(3), fsUntouched.get(1));
+  }
+
+  @Test
+  public void testBuildMergeClusteringGroup() throws Exception {
```

Comment:
This test flakes in my local runs. Can you please check?
<img width="1722" alt="Screenshot 2022-07-10 at 5 16 15 PM" src="https://user-images.githubusercontent.com/16440354/178143560-41b23226-64b5-40ee-b821-4a875b4815ca.png">

Reply:
Sure, I will rebase master and check all the tests